### PR TITLE
more explanation of UnionAll

### PR DESCRIPTION
**Pull request by StephenVavasis**
Explains the role of the position of the `where` keyword

### CODE REVIEW

**Review by JeffBezanson — APPROVED** [review]

File: doc/src/manual/types.md [hunk 1]
```diff
@@ -1002,6 +1002,20 @@ For example `Tuple{T,Array{S}} where S<:AbstractArray{T} where T<:Real` refers t
...
+```julia
+julia> const T1 = Array{Array{T,1} where T, 1}
+Array{Array{T,1} where T,1}
+
+julia> const T2 = Array{Array{T,1}, 1} where T
+Array{Array{T,1},1} where T
+```
+Type `T1` defines a 1-dimensional array of 1-dimensional arrays; each
+of the inner arrays consists of objects of the same type, but this type may vary from one inner array to the next.
+On the other hand, type `T2` defines a 1-dimensional array of 1-dimensional arrays all of whose inner arrays must have the
```
Comment (tkelman) [inline doc/src/manual/types.md]: there's going to be a whitespace failure here

Comment (Keno) [inline doc/src/manual/types.md]: We should have a quick-check CI for issues like this.

Comment (JeffBezanson) [inline doc/src/manual/types.md]: I will just state my opinion that trailing whitespace in the documentation is an absurd reason for a build to fail.

Comment (tkelman) [inline doc/src/manual/types.md]: Yes. Pick a travis-competitor service and we can play with it, make it a trivial fast test and have its status required for merging. Hopefully that will work without much of a queue. Or replace the whitespace check with an auto-fix script that we run once a week or so.

Comment (tkelman) [inline doc/src/manual/types.md]: Though even for uncontroversial doc additions like this one, there's no reason to rush merging - leaving it open for a bit lets people see and comment on it. Post-merge review tends to never get addressed and be forgotten.

**Pull request merged**